lib/http-client.js

```diff
@@ -49,6 +49,10 @@ module.exports = class {
     this.customHeaders = customHeaders || {};
   }
 
+  static getResponse(requestResponse) {
+    return requestResponse.__response; // eslint-disable-line no-underscore-dangle
```

Comment:
+1, good call on the API structure. If you want to make the internal implementation truly private and guarantee users will go through the method to get the property, you can do this easily by instantiating a private Symbol() for the httpClient on start and then setting and getting the property using the Symbol instance.

Reply:
I am hesitant because we'd need to find a polyfil, there are folks using this in the browser and Symbol isn't supported by any IE :(  -- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#Browser_compatibility